app/pages/patientdata/patientdata.js

```diff
@@ -526,6 +512,61 @@ export let PatientData = React.createClass({
       datetimeLocation: datetime
     });
   },
+
+  componentWillMount: function() {
+    this.doFetching(this.props);
+    this.doProcessing(this.props);
+    var params = this.props.queryParams;
+
+    if (!_.isEmpty(params)) {
+      var prefs = _.cloneDeep(this.state.chartPrefs);
+      prefs.bolusRatio = params.dynamicCarbs ? 0.5 : 0.35;
+      prefs.dynamicCarbs = params.dynamicCarbs;
+      this.setState({
+        chartPrefs: prefs
+      });
+    }
+  },
+
+  componentWillUnmount: function() {
+    this.props.clearPatientData(this.props.currentPatientInViewId);
+  },
+
+  componentWillReceiveProps: function(nextProps) {
+    var userId = this.props.currentPatientInViewId;
+    var currentPatientData = _.get(this.props, ['patientDataMap', userId], null);
+    
+    var nextPatientData = _.get(nextProps, ['patientDataMap', userId], null);
+    
+    if (!currentPatientData && nextPatientData) {
+      this.doProcessing(nextProps);
+    }
+  },
+
+  doProcessing: function(nextProps) {
+    // I realised the one source of truth we can rely on here to ensure we show the correct patient
+    // is to refer to the userId specified in the route! What do you think @jana?
+    var userId = _.get(this.props, 'routeParams.id', null);
```

Comment:
Now that you've mapped this is `currentPatientInViewId` use that? And then the next bit can also be reduced to something like `_.get(nextProps, ['patientDataMap', this.props.currentPatientInViewId || ''], null)`...